code/modules/rykka_puzzle_wips/reactor_defense.dm

```diff
@@ -0,0 +1,319 @@
+			update_icon()
+	return
+
+/obj/structure/damaged_reactor/bullet_act(var/obj/item/projectile/Proj) // A bullet hit us
+
+	var/proj_damage = Proj.get_structure_damage()
+	if(!proj_damage)
+		return
+
+	..()
+	take_damage(proj_damage) // Take damage according to the bullet's damage value.
+	return
+
+/obj/structure/damaged_reactor/blob_act() // In the unlikely event a blob is attacking the reactor, deal a flat 50 damage to it.
+	take_damage(50)
+	
+/obj/structure/damaged_reactor/hitby(AM as mob|obj) // I forget precisely what the fuck this is for but w/e.
+	..()
+	visible_message("<span class='danger'>\The [src] was hit by [AM].</span>")
+	var/tforce = 0
+	if(ismob(AM))
+		tforce = 40
+	else if(isobj(AM))
+		var/obj/item/I = AM
+		tforce = I.throwforce
+	take_damage(tforce)
+
+/obj/structure/damaged_reactor/attack_generic(var/mob/user, var/damage) // Are simplemobs trying to attack the reactor?
+	user.setClickCooldown(user.get_attack_speed()) // Control how fast they can attack/click
+	if(!damage) // If we're not doing damage for some reason, return.
+		return
+	if(damage >= STRUCTURE_MIN_DAMAGE_THRESHOLD) // Is the damage over the minimum required threshold? Deal damage.
+		visible_message("<span class='danger'>[user] tries to slice into \the [src] and does a little damage to the casing!</span>")
+		take_damage(damage)
+	else // Otherwise, they do nothing!
+		visible_message("<span class='notice'>\The [user] doesn't appear to do anything to \the [src]!</span>")
+	user.do_attack_animation(src)
+	return 1
+
+/obj/structure/damaged_reactor/attackby(obj/item/W as obj, mob/user as mob) // Object Interactions and more typecasting!
+	if(!istype(W)) // This handles an edge case scenario.
+		return
+
+	// Slamming the reactor with something/someone.
+	if(istype(W, /obj/item/weapon/grab) && get_dist(src,user)<2)
+		var/obj/item/weapon/grab/G = W
+		if(istype(G.affecting,/mob/living))
+			var/mob/living/M = G.affecting
+			var/state = G.state
+			qdel(W)	// gotta delete it here because if reactor breaks, it won't get deleted - not sure if this is needed since the reactor *can't* break.
+			switch (state)
+				if(1)
+					M.visible_message("<span class='warning'>[user] slams [M] against \the [src]!</span>")
+					M.apply_damage(7)
+					hit(10)
+				if(2)
+					M.visible_message("<span class='danger'>[user] bashes [M] against \the [src]!</span>")
+					if (prob(50))
+						M.Weaken(1)
+					M.apply_damage(10)
+					hit(25)
+				if(3)
+					M.visible_message("<span class='danger'><big>[user] crushes [M] against \the [src]!</big></span>")
+					M.Weaken(5)
+					M.apply_damage(20)
+					hit(50)
+			return
+	
+	if(W.flags & NOBLUDGEON) 
+		return
+	
+	else // A grab isn't the thing hitting us! Therefore, we receive damage based on the tool striking us!
+		user.setClickCooldown(user.get_attack_speed(W))
+		if(W.damtype == BRUTE || W.damtype == BURN)

```

Comment:
Probably should check `W.force` here or when you check for `NOBLUDGEON`.

Reply:
Check if it's actually set, or?

Reply:
Some items have force set to 0 to represent item being unable to deal damage. Might not want to go through with attack at all if it is like that.

Reply:
Yes, as Heroman said.

Reply:
Would a simple `	if(!W.force)
		return` work?

Reply:
`W.force <= 0` would also cover the (very strange) edge case of a negative force value.

Reply:
Added in, thank you <3